Modules/Core/Common/include/itkNeighborhoodAlgorithm.hxx

```diff
@@ -40,51 +39,49 @@ ImageBoundaryFacesCalculator<TImage>::Compute(const TImage & img, RegionType reg
   // and regionTOProcess (size = 5x5),
 
   Result result;
+
+  const RegionType bufferedRegion = img.GetBufferedRegion();
+
   // The portion of the regionToProcess that is outside of the image bufferedRegion
   // doesn't make sense. If the regionToProcess is completely outside of
   // the image bufferedRegion, return an empty (default-constructed) result.
-  if (!regionToProcess.Crop(img.GetBufferedRegion()))
+  if (!regionToProcess.Crop(bufferedRegion))
   {
     return result;
   }
 
-  const IndexType bStart = img.GetBufferedRegion().GetIndex();
-  const SizeType  bSize = img.GetBufferedRegion().GetSize();
+  const IndexType bStart = bufferedRegion.GetIndex();
+  const SizeType  bSize = bufferedRegion.GetSize();
   const IndexType rStart = regionToProcess.GetIndex();
   const SizeType  rSize = regionToProcess.GetSize();
 
-  IndexValueType overlapLow, overlapHigh;
-  IndexType      fStart; // Boundary, "face"
-  SizeType       fSize;  // region data.
-  RegionType     fRegion;
-  SizeType       nbSize = regionToProcess.GetSize();   // Non-boundary region
-  IndexType      nbStart = regionToProcess.GetIndex(); // data.
+  SizeType  nbSize = rSize;   // Non-boundary region
+  IndexType nbStart = rStart; // data.
 
   IndexType vrStart =
     rStart; // start index of variable processed region which has considered the boundary region in last direction
   SizeType vrSize =
     rSize; // size of variable processed region which has considered the boundary region in last direction
 
-  for (i = 0; i < ImageDimension; ++i)
+  for (unsigned int i = 0; i < ImageDimension; ++i)
   {
-    overlapLow = static_cast<IndexValueType>((rStart[i] - radius[i]) - bStart[i]);
+    IndexType fStart; // Boundary, "face"
+    SizeType  fSize;  // region data.
+
+    auto overlapLow = static_cast<IndexValueType>((rStart[i] - radius[i]) - bStart[i]);
     // image buffered region should be more than twice of the radius,
     // otherwise there would be overlap between two boundary regions.
     // in the case, we reduce upper boundary size to remove overlap.
-    if (bSize[i] > 2 * radius[i])
-    {
-      overlapHigh = static_cast<IndexValueType>((bStart[i] + bSize[i]) - (rStart[i] + rSize[i] + radius[i]));
-    }
-    else
-    {
-      overlapHigh = static_cast<IndexValueType>((bStart[i] + radius[i]) - (rStart[i] + rSize[i]));
-    }
 
-    if (overlapLow < 0)                    // out of bounds condition, define
-                                           // a region of
-    {                                      // iteration along this face
-      for (j = 0; j < ImageDimension; ++j) // define the starting index
-      {                                    // and size of the face region
+    auto overlapHigh = static_cast<IndexValueType>((bSize[i] > 2 * radius[i])
```

Comment:
I remember that someone was quite opposed to ternary operator. I thought it was you @blowekamp. Anyway, we can keep it in this instance instead of putting further effort into discussion.